apps/src/studio/Item.js

```diff
@@ -358,3 +444,37 @@ Item.prototype.moveToNextPosition = function () {
   this.x = next.x;
   this.y = next.y;
 };
+
+/**
+ * Mark that we're colliding with object represented by key.
+ * Here, override base implemention to special on-collision logic for certain
+ * item classes.
+ * @param key A unique key representing the object we're colliding with
+ * @returns {boolean} True if collision is started, false if we're already colliding
+ * @override
+ */
+Item.prototype.startCollision = function (key) {
+  var newCollisionStarted = Item.superPrototype.startCollision.call(this, key);
+  if (newCollisionStarted) {
+    if (this.className === 'hazard') {
+      var actor = Studio.sprite[key];
+      if (actor) {
+        Studio.sprite[key].addAction(new spriteActions.FadeActor(2000));
```

Comment:
Not for now, but I think we should consider how to expose these actions as blocks. Once you are in the freeplay level, it seems like students should be able to trigger some of these animations as a result of checking for specific collisions or other events. We've got some cool stuff in here now, but it isn't really exposed in the freeplay case.